control-plane/catalog/to-consul/resource.go

```diff
@@ -641,6 +667,24 @@ func (t *ServiceResource) generateRegistrations(key string) {
 	}
 }
 
+// Sets the passing service weight
+func (t *ServiceResource) setServiceWeight(weight string, appsCount int, r consulapi.CatalogRegistration) error {
```

Comment:
The caller need of be type `*ServiceResource` to use this function because it isn't getting used anywhere inside it. I would recommend to just have it as `func setServiceWeight` and move this function to the bottom of the file.